### ISSUE TITLE
IAE: object is not an instance of declaring class when loading some Gradle projects

### ISSUE THREAD

**Issue by lkishalmi** [apache/netbeans#4475]
### Apache NetBeans version

Apache NetBeans 15 release candidate

### What happened

Just opened and closed some project groups when I got the following exception:
```
java.lang.IllegalArgumentException: object is not an instance of declaring class
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.netbeans.modules.gradle.loaders.LegacyProjectLoader.getLocation(LegacyProjectLoader.java:332)
	at org.netbeans.modules.gradle.loaders.LegacyProjectLoader.createReport(LegacyProjectLoader.java:375)
	at org.netbeans.modules.gradle.loaders.LegacyProjectLoader.exceptionsToProblems(LegacyProjectLoader.java:314)
	at org.netbeans.modules.gradle.loaders.LegacyProjectLoader.loadGradleProject(LegacyProjectLoader.java:243)
	at org.netbeans.modules.gradle.loaders.LegacyProjectLoader.access$200(LegacyProjectLoader.java:89)
[catch] at org.netbeans.modules.gradle.loaders.LegacyProjectLoader$ProjectLoaderTask.call(LegacyProjectLoader.java:574)
	at org.netbeans.modules.gradle.loaders.LegacyProjectLoader$ProjectLoaderTask.call(LegacyProjectLoader.java:543)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at org.openide.util.RequestProcessor$Task.run(RequestProcessor.java:1418)
	at org.netbeans.modules.openide.util.GlobalLookup.execute(GlobalLookup.java:45)
	at org.openide.util.lookup.Lookups.executeWith(Lookups.java:278)
	at org.openide.util.RequestProcessor$Processor.run(RequestProcessor.java:2033)
```

Souldn't we do some instance checks before calling methods getLocation and getLineNumber? If that's not possible, probably we could just swallow this exception.

### How to reproduce

Well, I wish I could safely reproduce this one.

### Did this work correctly in an earlier version?

No / Don't know

### Operating System

Ubuntu 22.04

### JDK

JDK17

### Apache NetBeans packaging

Apache NetBeans Snap Package

### Anything else

Well, this one is not a blocker for NetBeans 15.

### Are you willing to submit a pull request?

Yes

### Code of Conduct

Yes

**Comment by neilcsmith-net** [comment]
@lkishalmi is this still valid?

**Comment by lkishalmi** [comment]
I think yes, though it is not a high priority issue as it does rarely happen. Moving to NB17.

**Comment by neilwightman** [comment]
This is happening to me every time I try to open a specific project.   Is there any work around for this bug?  I tried reinstalling netbeans 16, removing all ~/.netbeans/xxx directories and all nb related files in the project.   I can't figure out why netbeans was working fine before I had some food and now it's not working at all.  I even tried rebooting my computer.   

I have tried everything I can think of, but I'm will to try anything to help track down this issue.

I also built netbeans from source, and it happens with that version too.
